socket/socket.c

```diff
@@ -0,0 +1,427 @@
+#include <lua/lauxlib.h>
+#include <lua/lua.h>
+#include <lua/lualib.h>
+
+#include <linux/errno.h>
+#include <linux/in.h>
+#include <linux/inet.h>
+#include <linux/socket.h>
+#include <linux/string.h>
+#include <linux/unistd.h>
+#include <net/sock.h>
+
+#ifdef LUANTICK_DATA
+#include <luadata.h>
+#endif
+
+#include "enums.h"
+
+#define LUA_SOCKET "luasocket"
+#define LUA_SOCKET_MAXBUFFER 4096
+
+typedef struct socket *sock_t;
+
+extern const char *inet_ntop(int af, const void *src, char *dst, int size);
+extern int inet_pton(int af, const char *src, void *dst);
+
+lua_Integer lua_optfieldinteger(lua_State *L, int idx, const char *k, int def)
+{
+	int isnum;
+	lua_Integer d;
```

Comment:
it's a matter of code style, but we usually declare variables and then break the line, like:
```C
void foo(void)
{
   int a;

   if (...)
}
```